src/Kernel/Context.class.st

```diff
@@ -2257,19 +2257,25 @@ Context >> tryPrimitiveFor: aMethod receiver: aReceiver args: arguments [
 { #category : #'special context access' }
 Context >> unwindBlock [
 	"unwindContext only. access temporaries from BlockClosure>>#ensure: and BlockClosure>>#ifCurtailed:"
-	^self tempAt: 1
+
+	"The handler is in the first temporary"
+	^ self tempAt: self numArgs + 1
 ]
 
 { #category : #'special context access' }
 Context >> unwindComplete [
 	"unwindContext only. access temporaries from BlockClosure>>#ensure: and BlockClosure>>#ifCurtailed:"
-	^self tempAt: 2
+
+	"The complete flag is in the antepenultimate temporary"
+	^ self tempAt: self numTemps - 1
 ]
```

Comment:
Would it be better to reserve second temp for complete var? It sounds more intuitive that all reserved temps are in same order. 
Now it's a bit strange to reserve first temp and antepenultimate temp. Proxy with many temps will confuse

Reply:
you mean having `^ self tempAt: self numArgs + 2` ?

I'm ok with that :)